src/lib.rs

```diff
@@ -155,16 +197,25 @@ pub struct CanMessage {
 
 
 impl<'a> Panda<'a>  {
-    pub fn new(context: &'a libusb::Context, timeout: Duration) -> Panda<'a>  {
+    pub fn new(context: &'a libusb::Context, timeout: Duration) -> Self {
         let device = context.open_device_with_vid_pid(0xbbaa, 0xddcc).unwrap();
 
-        Panda {
+        let mut panda = Self {
             device,
-            timeout
-        }
+            timeout,
+            packet_versions: PacketVersions {
+                health_version: 0,
+                can_version: 0,
+            },
+        };
+        panda.packet_versions = panda.get_packet_versions().unwrap();
```

Comment:
Maybe a function  like `update_packet_versions` can just modify the object in place? Not sure if that's much cleaner.

`Panda::new()` should return a `Result<Panda, libusb::Error>` to get rid of the unwraps, but that's for another time.

Reply:
I've made some changes... We could create a constructor for PacketVersions and read the endpoint there? Then we don't have to update the struct after creating it.